src/Microsoft.PowerShell.Commands.Utility/commands/utility/Write-Object.cs

```diff
@@ -55,10 +55,8 @@ protected override void ProcessRecord()
             {
                 enumerate = false;
             }
-            foreach (PSObject inputObject in _inputObjects) // compensate for ValueFromRemainingArguments
-            {
-                WriteObject(inputObject, enumerate);
-            }
+
+            WriteObject(_inputObjects, enumerate);
```

Comment:
>In case of a b c, I gave you 3 items, so 3 items should be returned by Write-Output
> in case of a,b,c, I gave you 1 item,

I disagree with that.  `ValueFromRemainingArguments` is just syntax sugar, similar to a `params` parameter in C#.  You gave 3 objects in both versions of the command.